### PR TITLE
Fixes #16774: rl allow splitted global variables declaration

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/16774

### CODE REVIEW

File: rudder-lang/src/ast/context.rs [hunk 1]
```diff
@@ -43,9 +43,8 @@ impl<'src> VarContext<'src> {
         self.variables.iter()
     }
 
-    /// Create a variable in the context.
-    /// Used by dedicated variable creation method for each kind.
-    fn new_var(
+    /// Checks whether a variable already exists in a given scope
+    fn new_var_checker(
```
Comment (peckpeck) [inline rudder-lang/src/ast/context.rs]: new var checker means that you will return a checker, which is not what is done

File: rudder-lang/src/parser.rs [hunk 2]
```diff
@@ -600,9 +600,24 @@ fn pstruct(i: PInput) -> PResult<HashMap<String, PValue>> {
     )(i)
 }
 
+/// Alternative version of pstruct based on "." rather than braces.
+/// Used for the agent
+/// A struct is stored in a HashMap
+// fn pstruct_agent(i: PInput) -> PResult<HashMap<String, PValue>> {
```
Comment (peckpeck) [inline rudder-lang/src/parser.rs]: Why is this here ?

Comment (gpoblon) [reply rudder-lang/src/parser.rs]: I have to manually remove it from every curent PR since it is the same code i rebased on last week

**Comment by gpoblon** [comment]
cleaned comments and rebased on master, should be mergable

**Comment by peckpeck** [comment]
oops it seems you removed everything, this PR seems empty now

**Comment by gpoblon** [comment]
PR rebased

**Comment by gpoblon** [comment]
PR rebased

File: rudder-lang/src/ast.rs [hunk 3]
```diff
@@ -20,6 +20,7 @@ use self::value::Value;
 // TODO check that parameter type match parameter default
 // TODO check state call compatibility
 
+// Ptye -> pvalue
```
Comment (peckpeck) [inline rudder-lang/src/ast.rs]: what is this ?

File: rudder-lang/src/ast/context.rs [hunk 4]
```diff
@@ -82,20 +82,62 @@ impl<'src> VarContext<'src> {
...
         value: Option<Token<'src>>,
     ) -> Result<()> {
-        self.new_var(upper_context, name)?;
+        self.check_new_var(upper_context, name)?;
         self.variables.insert(name, VarKind::Enum(enum1, value));
         Ok(())
     }
 
-    /// Create a generic variable in tis context.
+    fn push_or_err_rec(name: Token<'src>, stored_ctx: &mut Value<'src>, new_var: &Value<'src>) -> Result<()> {
```
Comment (peckpeck) [inline rudder-lang/src/ast/context.rs]: This function name is a bit meaningless, can you comment it and rename it ?

**Comment by gpoblon** [comment]
PR updated with a new commit

File: rudder-lang/src/ast/context.rs (hunk 4 above)
Comment (gpoblon) [reply rudder-lang/src/ast/context.rs]: improved it

Comment (gpoblon) [reply rudder-lang/src/ast/context.rs]: improved it

File: rudder-lang/src/ast.rs (hunk 3 above)
Comment (gpoblon) [reply rudder-lang/src/ast.rs]: good question, deleted it

**Comment by amousset** [comment]
OK, squash merging this PR

**Comment by peckpeck** [comment]
OK, merging this PR
